### PR TITLE
Add Status.Annotations to the conversion workflow

### PR DESCRIPTION
**Pull request by vagababov**
/assign @dprotaso @vaikas

### CODE REVIEW

File: apis/duck/v1/status_types.go [hunk 1]
```diff
@@ -93,6 +94,8 @@ func (s *Status) GetCondition(t apis.ConditionType) *apis.Condition {
 // return true the condition type will be copied to the sink
 func (source *Status) ConvertTo(ctx context.Context, sink *Status, predicates ...func(apis.ConditionType) bool) {
 	sink.ObservedGeneration = source.ObservedGeneration
+	// This will deep copy the map.
+	sink.Annotations = kmeta.UnionMaps(source.Annotations, nil)
```
Comment (dprotaso) [inline apis/duck/v1/status_types.go:98]: if only union maps was a variadic ;)

Comment (vagababov) [reply apis/duck/v1/status_types.go:98]: I can do that, actually with backward compatibility, I guess.
But again different exercise :D

**Comment by vaikas** [comment]
/lgtm
/approve

Thanks!!!

**Pull request merged**